pkg/template/servicebroker/provision.go

```diff
@@ -240,19 +286,16 @@ func (b *Broker) Provision(instanceID string, preq *api.ProvisionRequest) *api.R
 		return api.BadRequest(kerrors.NewNotFound(templateapi.Resource("templates"), preq.ServiceID))
 	}
 
-	// TODO: enable SAR for template - at the moment I think this doesn't work
-	// properly because group information isn't populated in u.
-	/*
-		if err := util.Authorize(b.kc.Authorization().SubjectAccessReviews(), u, &authorization.ResourceAttributes{
-			Namespace: template.Namespace,
-			Verb:      "get",
-			Group:     templateapi.GroupName,
-			Resource:  "templates",
-			Name:      template.Name,
-		}); err != nil {
-			return api.Forbidden(err)
-		}
-	*/
+	// with groups in the user.Info vs. the username only form of auth, we can SAR for get access on template resources
+	if err := util.Authorize(b.kc.Authorization().SubjectAccessReviews(), u, &authorization.ResourceAttributes{
```

Comment:
OK, I have confirmed that this SAR will pass when I explicitly use `add-role-to-user` for the `openshift` namespace in the extended tests.  Removing those calls, and this SAR fails.

I've added comments detailing this around this call, as well as the `add-role-to-user` calls in the extended tests.

At this point, I think we should leave this as is until the service catalog changes arrive, and it is actually authenticating, and we are using the user.Info from the api.Context instead of the bind/provision request parameter map.  At that time, when all these //TODO workaround comments are cleaned up, we remove the `add-role-to-user` calls in the extended tests, and verify this SAR still passes / fails as appropriate.